### PR TITLE
Feature/descw 2677 devcontainer

### PR DESCRIPTION
**Pull request by mhaswell-bcgov**
Can someone test this on their machine?
Steps to using this devcontainer config:

1. Install the [Dev Containers VSCode extension](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-containers).
2. View > Command Palette... > Dev Containers: Reopen in Container
3. VSCode should relaunch and the terminal should look like `vscode ➜ /workspaces/naad-connector`
4. From that terminal, run `composer test-coverage`. It should succeed and generate a coverage report in `coverage/`. 

Certain things are hardcoded (devcontainer.env and the PHPMyAdmin environment variables in .devcontainer/docker-compose.yaml) that shouldn't be in this configuration but I'm just wanting to see if this is working for everyone.

### CODE REVIEW

**Comment by robrien-bcgov** [comment]
It's prompting me for two things:

1. Add configuration to workspace / Add configuration to user data folder

2. How would you like to create your container configuration?
- from a predefined container configuration template
- from Dockerfile
- from docker-compose.yaml


The Dockerfile option didn't work
The docker-compose.yaml option asked me which service to run

I will try just using a predefined container configuration template instead.

**Comment by robrien-bcgov** [comment]
Had an error when trying to stand up the devcontainer:


```shell
01e39d6cd9.yml up -d
[+] Running 3/3
 ✘ database-migrate Error pull access denied fo...                         0.7s 
 ✘ phpmyadmin Error       context canceled                                 0.7s 
 ✘ mariadb Error          context canceled                                 0.7s 
Error response from daemon: pull access denied for bcgovgdx/naad-app, repository does not exist or may require 'docker login'
[2476 ms] Error: Command failed: docker compose --project-name naad-connector_devcontainer -f /Users/robrien/Development/naad-connector/.devcontainer/docker-compose.yml -f /Users/robrien/Library/Application Support/Code/User/globalStorage/ms-vscode-remote.remote-containers/data/docker-compose/docker-compose.devcontainer.build-1732572929768.yml -f /Users/robrien/Library/Application Support/Code/User/globalStorage/ms-vscode-remote.remote-containers/data/docker-compose/docker-compose.devcontainer.containerFeatures-1732572930738-8d8622c6-3546-4b61-b23b-4701e39d6cd9.yml up -d
[2476 ms]     at ftA (/Users/robrien/.vscode/extensions/ms-vscode-remote.remote-containers-0.388.0/dist/spec-node/devContainersSpecCLI.js:430:3355)
[2476 ms]     at async htA (/Users/robrien/.vscode/extensions/ms-vscode-remote.remote-containers-0.388.0/dist/spec-node/devContainersSpecCLI.js:410:3496)
[2476 ms]     at async TtA (/Users/robrien/.vscode/extensions/ms-vscode-remote.remote-containers-0.388.0/dist/spec-node/devContainersSpecCLI.js:482:4021)
[2476 ms]     at async iB (/Users/robrien/.vscode/extensions/ms-vscode-remote.remote-containers-0.388.0/dist/spec-node/devContainersSpecCLI.js:482:4963)
[2476 ms]     at async wrA (/Users/robrien/.vscode/extensions/ms-vscode-remote.remote-containers-0.388.0/dist/spec-node/devContainersSpecCLI.js:663:203)
[2476 ms]     at async DrA (/Users/robrien/.vscode/extensions/ms-vscode-remote.remote-containers-0.388.0/dist/spec-node/devContainersSpecCLI.js:662:14830)
[2476 ms]     at async /Users/robrien/.vscode/extensions/ms-vscode-remote.remote-containers-0.388.0/dist/spec-node/devContainersSpecCLI.js:482:1190
[2479 ms] Exit code 1
[2482 ms] Command failed: /Applications/Visual Studio Code.app/Contents/Frameworks/Code Helper (Plugin).app/Contents/MacOS/Code Helper (Plugin) /Users/robrien/.vscode/extensions/ms-vscode-remote.remote-containers-0.388.0/dist/spec-node/devContainersSpecCLI.js up --user-data-folder /Users/robrien/Library/Application Support/Code/User/globalStorage/ms-vscode-remote.remote-containers/data --container-session-data-folder /tmp/devcontainers-40457ea8-6de0-4320-8bd9-6acb3b56fd3c1732572928561 --workspace-folder /Users/robrien/Development/naad-connector --workspace-mount-consistency cached --gpu-availability detect --id-label devcontainer.local_folder=/Users/robrien/Development/naad-connector --id-label devcontainer.config_file=/Users/robrien/Development/naad-connector/.devcontainer/devcontainer.json --log-level debug --log-format json --config /Users/robrien/Development/naad-connector/.devcontainer/devcontainer.json --default-user-env-probe loginInteractiveShell --mount type=volume,source=vscode,target=/vscode,external=true --skip-post-create --update-remote-user-uid-default on --mount-workspace-git-root --include-configuration --include-merged-configuration
[2482 ms] Exit code 1
```

**Comment by mhaswell-bcgov** [comment]
> I got an error - I guess something about docker login, or credentials for pulling the images from the repository.
> 
> ```shell
> [1659 ms] Start: Run: docker inspect --type image naad-connector_devcontainer-app
> [1688 ms] Docker Compose override file for creating container:
> services:
>   'app':
>     entrypoint: ["/bin/sh", "-c", "echo Container started\n
> trap \"exit 0\" 15\n
> \n
> exec \"$$@\"\n
> while sleep 1 & wait $$!; do :; done", "-", "docker-php-entrypoint"]
>     labels:
>       - 'devcontainer.local_folder=/Users/robrien/Development/naad-connector'
>       - 'devcontainer.config_file=/Users/robrien/Development/naad-connector/.devcontainer/devcontainer.json'
>     volumes:
>       - vscode:/vscode
> volumes:
>   
>   vscode:
>     external: true
> [1689 ms] Writing docker-compose.devcontainer.containerFeatures-1732572930738-8d8622c6-3546-4b61-b23b-4701e39d6cd9.yml to /Users/robrien/Library/Application Support/Code/User/globalStorage/ms-vscode-remote.remote-containers/data/docker-compose
> [1689 ms] Start: Run: docker compose --project-name naad-connector_devcontainer -f /Users/robrien/Development/naad-connector/.devcontainer/docker-compose.yml -f /Users/robrien/Library/Application Support/Code/User/globalStorage/ms-vscode-remote.remote-containers/data/docker-compose/docker-compose.devcontainer.build-1732572929768.yml -f /Users/robrien/Library/Application Support/Code/User/globalStorage/ms-vscode-remote.remote-containers/data/docker-compose/docker-compose.devcontainer.containerFeatures-1732572930738-8d8622c6-3546-4b61-b23b-4701e39d6cd9.yml up -d
> [+] Running 3/3
>  ✘ database-migrate Error pull access denied fo...                         0.7s 
>  ✘ phpmyadmin Error       context canceled                                 0.7s 
>  ✘ mariadb Error          context canceled                                 0.7s 
> Error response from daemon: pull access denied for bcgovgdx/naad-app, repository does not exist or may require 'docker login'
> [2476 ms] Error: Command failed: docker compose --project-name naad-connector_devcontainer -f /Users/robrien/Development/naad-connector/.devcontainer/docker-compose.yml -f /Users/robrien/Library/Application Support/Code/User/globalStorage/ms-vscode-remote.remote-containers/data/docker-compose/docker-compose.devcontainer.build-1732572929768.yml -f /Users/robrien/Library/Application Support/Code/User/globalStorage/ms-vscode-remote.remote-containers/data/docker-compose/docker-compose.devcontainer.containerFeatures-1732572930738-8d8622c6-3546-4b61-b23b-4701e39d6cd9.yml up -d
> [2476 ms]     at ftA (/Users/robrien/.vscode/extensions/ms-vscode-remote.remote-containers-0.388.0/dist/spec-node/devContainersSpecCLI.js:430:3355)
> [2476 ms]     at async htA (/Users/robrien/.vscode/extensions/ms-vscode-remote.remote-containers-0.388.0/dist/spec-node/devContainersSpecCLI.js:410:3496)
> [2476 ms]     at async TtA (/Users/robrien/.vscode/extensions/ms-vscode-remote.remote-containers-0.388.0/dist/spec-node/devContainersSpecCLI.js:482:4021)
> [2476 ms]     at async iB (/Users/robrien/.vscode/extensions/ms-vscode-remote.remote-containers-0.388.0/dist/spec-node/devContainersSpecCLI.js:482:4963)
> [2476 ms]     at async wrA (/Users/robrien/.vscode/extensions/ms-vscode-remote.remote-containers-0.388.0/dist/spec-node/devContainersSpecCLI.js:663:203)
> [2476 ms]     at async DrA (/Users/robrien/.vscode/extensions/ms-vscode-remote.remote-containers-0.388.0/dist/spec-node/devContainersSpecCLI.js:662:14830)
> [2476 ms]     at async /Users/robrien/.vscode/extensions/ms-vscode-remote.remote-containers-0.388.0/dist/spec-node/devContainersSpecCLI.js:482:1190
> [2479 ms] Exit code 1
> [2482 ms] Command failed: /Applications/Visual Studio Code.app/Contents/Frameworks/Code Helper (Plugin).app/Contents/MacOS/Code Helper (Plugin) /Users/robrien/.vscode/extensions/ms-vscode-remote.remote-containers-0.388.0/dist/spec-node/devContainersSpecCLI.js up --user-data-folder /Users/robrien/Library/Application Support/Code/User/globalStorage/ms-vscode-remote.remote-containers/data --container-session-data-folder /tmp/devcontainers-40457ea8-6de0-4320-8bd9-6acb3b56fd3c1732572928561 --workspace-folder /Users/robrien/Development/naad-connector --workspace-mount-consistency cached --gpu-availability detect --id-label devcontainer.local_folder=/Users/robrien/Development/naad-connector --id-label devcontainer.config_file=/Users/robrien/Development/naad-connector/.devcontainer/devcontainer.json --log-level debug --log-format json --config /Users/robrien/Development/naad-connector/.devcontainer/devcontainer.json --default-user-env-probe loginInteractiveShell --mount type=volume,source=vscode,target=/vscode,external=true --skip-post-create --update-remote-user-uid-default on --mount-workspace-git-root --include-configuration --include-merged-configuration
> [2482 ms] Exit code 1
> ```

@robrien-bcgov I think this should fix the issues with the database-migrate pod (because I removed it).

File: .devcontainer/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    && apt-get install -y libzip-dev \
+    && apt-get clean -y && rm -rf /var/lib/apt/lists/*
+
+# Install php-mysql driver
+RUN docker-php-ext-install mysqli pdo pdo_mysql zip sockets \
+    && docker-php-ext-enable sockets
+
+# [Optional] Uncomment this section to install additional OS packages.
+# RUN apt-get update && export DEBIAN_FRONTEND=noninteractive \
+#     && apt-get -y install --no-install-recommends <your-package-list-here>
```
Comment (robrien-bcgov) [inline .devcontainer/Dockerfile:15]: it runs for me - not sure how to look at logs and stuff, but I like the idea of not having to rebuild for every edit.

**Review by robrien-bcgov — DISMISSED** [review]

File: .devcontainer/Dockerfile (hunk 1 above)
Comment (mhaswell-bcgov) [reply .devcontainer/Dockerfile:15]: The application logs should show up in the terminal you start it from in VSCode. For the other pods' logs I've been looking at docker desktop:
image

**Review by robrien-bcgov — APPROVED** [review]

**Pull request merged**